### PR TITLE
Update to version 0.14.0

### PR DESCRIPTION
**Pull request by kszucs**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -34,8 +34,11 @@ requirements:
     - arrow-cpp {{ version }}
 
   run:
+    - glog
+    - snappy
     - boost-cpp
     - python
+    - setuptools
     - {{ pin_compatible('numpy', lower_bound='1.14') }}
     - pandas
```
Comment (kszucs) [inline recipe/meta.yaml]: Pandas is not a hard dependency anymore, but I suggest to remove it in the next release.

**Comment by kszucs** [comment]
@xhochy @pitrou please review

**Comment by kszucs** [comment]
@conda-forge-admin, please rerender

File: recipe/build.sh [hunk 2]
```diff
@@ -9,6 +9,13 @@ export PARQUET_HOME=$PREFIX
 export SETUPTOOLS_SCM_PRETEND_VERSION=$PKG_VERSION
 
 cd python
+
 $PYTHON setup.py \
-        build_ext --build-type=release --with-parquet --with-plasma --with-orc \
-        install --single-version-externally-managed --record=record.txt
+        build_ext --build-type=release \
+                  --with-flight \
```
Comment (xhochy) [inline recipe/build.sh]: Why not add these to the Windows script as well?

File: recipe/bld.bat [hunk 3]
```diff
@@ -14,9 +14,12 @@ copy /Y "%SRC_DIR%\cpp\cmake_modules\FindPythonLibsNew.cmake" cmake_modules\
...
 SET ARROW_HOME=%LIBRARY_PREFIX%
 SET SETUPTOOLS_SCM_PRETEND_VERSION=%PKG_VERSION%
-SET "PYARROW_CMAKE_GENERATOR=NMake Makefiles"
+SET PYARROW_WITH_FLIGHT=1
+SET PYARROW_WITH_GANDIVA=1
+SET PYARROW_WITH_PARQUET=1
 "%PYTHON%" setup.py ^
-           build_ext --build-type=release --with-parquet ^
-           install --single-version-externally-managed --record=record.txt
+           build_ext --build-type=release ^
```
Comment (pitrou) [inline recipe/bld.bat]: Let's use `SET PYARROW_BUILD_TYPE=release` instead.

File: recipe/build.sh [hunk 4]
```diff
@@ -9,6 +9,13 @@ export PARQUET_HOME=$PREFIX
...
+                  --with-orc \
+                  --with-plasma \
+                  --with-parquet \
+                  --with-gandiva \
```
Comment (pitrou) [inline recipe/build.sh]: Let's use environment variables here as well. Passing options on the `setup.py` command-line is a bit cumbersome and error-prone.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -34,6 +34,8 @@ requirements:
     - arrow-cpp {{ version }}
 
   run:
+    - glog
```
Comment (pitrou) [inline recipe/meta.yaml]: Ideally those lists would be alphabetically-ordered...

**Comment by kszucs** [comment]
Do We support `PYARROW_WITH_ORC` on windows?

**Comment by kszucs** [comment]
@pitrou @xhochy updated

**Pull request merged**